src/overlays/actors/ovl_Shot_Sun/z_shot_sun.c

```diff
@@ -32,27 +30,156 @@ const ActorInit Shot_Sun_InitVars = {
-    { ELEMTYPE_UNK0, { 0x00000000, 0x00, 0x00 }, { 0x00000020, 0x00, 0x00 }, TOUCH_NONE | TOUCH_SFX_NORMAL, BUMP_ON, OCELEM_ON, },
+static ColliderCylinderInit sCylinderInit = {
+    {
+        COLTYPE_NONE,
+        AT_NONE,
+        AC_ON | AC_TYPE_PLAYER,
+        OC1_ON | OC1_TYPE_ALL,
+        OC2_TYPE_1,
+        COLSHAPE_CYLINDER,
+    },
+    {
+        ELEMTYPE_UNK0,
+        { 0x00000000, 0x00, 0x00 },
+        { 0x00000020, 0x00, 0x00 },
+        TOUCH_NONE | TOUCH_SFX_NORMAL,
+        BUMP_ON,
+        OCELEM_ON,
+    },
     { 30, 60, 0, { 0, 0, 0 } },
 };
 
-#endif
-
-extern ColliderCylinderInit D_80973BA0;
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Shot_Sun/ShotSun_Init.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Shot_Sun/ShotSun_Destroy.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Shot_Sun/func_80973740.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Shot_Sun/func_80973804.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Shot_Sun/func_809738D0.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Shot_Sun/func_80973960.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Shot_Sun/ShotSun_Update.s")
+void ShotSun_Init(Actor* thisx, PlayState* play) {
+    s32 pad;
+    ShotSun* this = THIS;
+
+    if ((SHOTSUN_GET_FF(thisx) == SHOTSUN_FAIRY_SPAWNER_SUNS) ||
+        (SHOTSUN_GET_FF(thisx) == SHOTSUN_FAIRY_SPAWNER_STORMS)) {
+        this->unk_19C = 0;
+        this->actor.flags |= ACTOR_FLAG_10;
+        this->actor.flags |= ACTOR_FLAG_2000000;
+        this->actionFunc = func_809738D0;
+        this->actor.flags |= ACTOR_FLAG_8000000;
+    } else {
+        Collider_InitCylinder(play, &this->collider);
+        Collider_SetCylinder(play, &this->collider, &this->actor, &sCylinderInit);
+        this->actionFunc = ShotSun_UpdateHyliaSun;
+        this->actor.flags &= ~ACTOR_FLAG_1;
+    }
+}
+
+void ShotSun_Destroy(Actor* thisx, PlayState* play) {
+    ShotSun* this = THIS;
+
+    if ((SHOTSUN_GET_FF(thisx) != SHOTSUN_FAIRY_SPAWNER_SUNS) &&
+        (SHOTSUN_GET_FF(thisx) != SHOTSUN_FAIRY_SPAWNER_STORMS)) {
+        Collider_DestroyCylinder(play, &this->collider);
+    }
+}
+
+void ShotSun_SpawnFairy(ShotSun* this, PlayState* play2) {
+    s32 params = SHOTSUN_GET_FF(&this->actor);
+    s32 fairyType = 0;
+    PlayState* play = play2;
+
+    if (this->timer > 0) {
+        this->timer--;
+    } else {
+        ActorCutscene_Stop(this->actor.cutscene);
+        switch (params) {
+            case SHOTSUN_FAIRY_SPAWNER_SUNS:
+                fairyType = 7;
+                break;
+            case SHOTSUN_FAIRY_SPAWNER_STORMS:
+                fairyType = 7;
+                break;
+        }
+        Actor_Spawn(&play->actorCtx, play, ACTOR_EN_ELF, this->actor.home.pos.x, this->actor.home.pos.y,
+                    this->actor.home.pos.z, 0, 0, 0, fairyType);
```

Comment:
We should be using the en_elf actor enum for fairyType

Reply:
Could just create a generic (something like `ENELF_TYPE_7`) enum in `z_en_elf.h`, that gets name/documented with `EnElf`